tests/units/feature_extraction/test_data.py

```diff
@@ -0,0 +1,173 @@
+                          index=[500] * 20, name="val"))]
+
+WIDE_TEST_DATA_EXPECTED_TUPLES = \
+    [(10, 'a', pd.Series([11, 9, 67, 45, 30, 58, 62, 19, 56, 29, 0, 27, 36,
+                          43, 33, 2, 24, 71, 41, 28],
+                         index=list(range(20)), name="a")),
+     (10, 'b', pd.Series([50, 40, 39, 7, 53, 23, 16, 37, 66, 38, 6, 47, 3,
+                          61, 44, 42, 78, 31, 21, 55],
+                         index=list(range(20)), name="b")),
+     (500, 'a', pd.Series([15, 35, 25, 32, 69, 65, 70, 64, 51, 46, 5, 77,
+                           26, 73, 76, 75, 72, 74, 10, 57],
+                          index=list(range(20, 40)), name="a")),
+     (500, 'b', pd.Series([4, 14, 68, 22, 18, 52, 54, 60, 79, 12, 49, 63,
+                           8, 59, 1, 13, 20, 17, 48, 34],
+                          index=list(range(20, 40)), name="b"))]
+
+
+class DataAdapterTestCase(DataTestCase):
+
+    def test_long_tsframe(self):
+        df = self.create_test_data_sample()
+        data = LongTsFrameAdapter(df, "id", "kind", "val", "sort")
+
+        self.assert_tsdata(data, TEST_DATA_EXPECTED_TUPLES)
+
+    def test_wide_tsframe(self):
+        df = self.create_test_data_sample_wide()
+        data = WideTsFrameAdapter(df, "id", "sort")
+
+        self.assert_tsdata(data, WIDE_TEST_DATA_EXPECTED_TUPLES)
+
+    def test_dict_tsframe(self):
+        df = {key: df for key, df in self.create_test_data_sample().groupby(["kind"])}
+        data = TsDictAdapter(df, "id", "val", "sort")
+
+        self.assert_tsdata(data, TEST_DATA_EXPECTED_TUPLES)
+
+    def assert_tsdata(self, data, expected):
+        self.assertEqual(len(data), len(expected))
+        self.assertEqual(sum(1 for _ in data), len(data))
+        self.assertEqual(sum(1 for _ in data.partition(1)), len(expected))
+        self.assertEqual((sum(sum(1 for _ in g) for g in data.partition(1))), len(data))
+        self.assert_data_chunk_object_equal(data, expected)
+
+    def assert_data_chunk_object_equal(self, result, expected):
+        dic_result = {str(x[0]) + "_" + str(x[1]): x[2] for x in result}
+        dic_expected = {str(x[0]) + "_" + str(x[1]): x[2] for x in expected}
+        for k in dic_result.keys():
+            pd.testing.assert_series_equal(dic_result[k], dic_expected[k])
+
+    def test_simple_data_sample_two_timeseries(self):
+        df = pd.DataFrame({"id": [10] * 4, "kind": ["a"] * 2 + ["b"] * 2, "val": [36, 71, 78, 37]})
+        df.set_index("id", drop=False, inplace=True)
+        df.index.name = None
+
+        result = to_tsdata(df, "id", "kind", "val")
+        expected = [(10, 'a', pd.Series([36, 71], index=[10] * 2, name="val")),
+                    (10, 'b', pd.Series([78, 37], index=[10] * 2, name="val"))]
+        self.assert_data_chunk_object_equal(result, expected)
+
+    def test_simple_data_sample_four_timeseries(self):
+        df = self.create_test_data_sample()
+        # todo: investigate the names that are given
```

Comment:
what does that mean?

Reply:
No idea :D

Just migrated some of your own tests: https://github.com/blue-yonder/tsfresh/blob/2a50ba7e34aacaa3c7fe62b6859e267186579308/tests/units/feature_extraction/test_extraction.py#L292